src/Engines/nativeEngine.ts

```diff
@@ -87,6 +87,43 @@ interface INativeEngine {
     readonly ALPHA_SCREENMODE: number;
 
     readonly homogeneousDepth: boolean;
+    readonly originBottomLeft: boolean;

```

Comment:
I might be thinking about this wrong, but it seems odd to me for JavaScript to know about and fix what I think of as a fundamentally native problem. Off the top of my head, this feels like a setting that should maybe be provided per framebuffer by the Graphics component, then leveraged where appropriate by the NativeEngine down in the C++ layer. Is the idea of screen origin location ever exposed to JavaScript by browsers?

Reply:
I agree YFlip is only a native problem. In practice, NativeShaderProcessor has to append the flip, so this boolean has to be exposed. Concerning the fullscreen pass, same, only a native problem and the pass 'should' be done on the c++ side. But, for maintenance reason, I don't want to maintain a precompiled shader version for all the platforms (build shaderc, have bgfx shader generated, bin2h, integrate shaders in the solution, ...). Just building shaderc and all its dependencies will blow up the compilation time on the ci. It's a trade off.

> Is the idea of screen origin location ever exposed to JavaScript by browsers?

Yes and no. There is an implicit difference between webgl and webgpu. Depending on the API you use, you have to flip or not.